src/api/argo-templates/base/sat-general-template-iuf.argo.yaml

```diff
@@ -0,0 +1,103 @@
+#
+# MIT License
+#
+# (C) Copyright 2022 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: sat-general
+spec:
+  entrypoint: main
+  templates:
+    ### Main Steps ###
+    - name: main
+      inputs:
+        parameters:
+          - name: auth_token
+          - name: global_params
+          - name: scriptContent
+      steps:
+        - - name:
+            template: sat-wrapper
+            arguments:
+              parameters:
+                - name: auth_token
+                  value: "{{inputs.parameters.auth_token}}"
+                - name: scriptContent
+                  value: "{{inputs.parameters.scriptContent}}"
+    ### Templates ###
+    ## sat-wrapper ##
+    - name: sat-wrapper
+      nodeSelector:
+        kubernetes.io/hostname: ncn-m001
+      tolerations:
+        - key: node-role.kubernetes.io/master
+          effect: NoSchedule
```

Comment:
Do we need the `nodeSelector` and `tolerations` here or is it more appropriate to have that on the Argo workflow template that uses this one (e.g. the workflow template for the `prepare_images` operation)?

I'm not even sure we have any reason why we need to be running only on ncn-m001. I don't believe the operations that use `sat` commands really need to use anything from the rbd mount, for example.

Reply:
I thought all IUF operations were supposed to be on ncn-m001. That is a good question regarding the calling templates... I am unsure hopefully I can test that out tomorrow on Frigg and circle back to this comment.

Reply:
I'm also unsure about that, I thought I remembered reading that in the IUF Development FAQ, though looking at it again, I also don't think there's anything specific to m001 that we'd be using here (at least in the IUF context). The things I can think of that require m001:
* rbd mount
* ssh keys configured for access to other mgmt nodes
* files on m001 filesystems

Bootprep only relies on the token, config, input files, and k8s config being present in the container, and the k8s API being accessible, and I think we're handling all of that with the workflow template scaffolding here.

Reply:
Yeah, we might actually need the bootprep files from their paths on the ncn-m001 local filesystem (or maybe the rbd mount). Let's just leave this for now, and if we can loosen it up later, great.